pkg/operator/rules_test.go

```diff
@@ -246,3 +251,101 @@ func TestGenerateGlobalRules(t *testing.T) {
 		})
 	}
 }
+
+func TestHasRules(t *testing.T) {
+	emptyRules := newFakeClientBuilder().Build()
+	hasRule := newFakeClientBuilder().WithObjects(&monitoringv1.Rules{}).Build()
+	errGettingRules := newFakeClientBuilder().WithInterceptorFuncs(interceptor.Funcs{
+		List: func(_ context.Context, _ client.WithWatch, _ client.ObjectList, _ ...client.ListOption) error {
+			return errors.New("error")
+		}}).Build()
+
+	type test struct {
+		client  client.Client
+		want    bool
+		wantErr bool
+	}
+
+	tests := map[string]test{
+		"no rules": {client: emptyRules, want: false},
+		"has rule": {client: hasRule, want: true},
+		"error":    {client: errGettingRules, want: false, wantErr: true},
+	}
+
+	for name, tc := range tests {
+		t.Run(name, func(t *testing.T) {
+			r := rulesReconciler{
+				client: tc.client,
+			}
+			got, err := r.hasRules(context.Background())
+			if got != tc.want {
+				t.Errorf("want: %t, got: %t", tc.want, got)
+			}
+			if err != nil && !tc.wantErr {
+				t.Errorf("Unexpected error: %s", err)
+			}
+		})
+	}
+}
+
+func TestScaleRuleConsumers(t *testing.T) {
+	var alertmanagerReplicas int32
+	alertManager := appsv1.StatefulSet{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "alertmanager",
+		},
+		Spec: appsv1.StatefulSetSpec{
+			Replicas: &alertmanagerReplicas,
+		},
+	}
+	var ruleEvaluatorReplicas int32
+	ruleEvaluator := appsv1.Deployment{
+		ObjectMeta: metav1.ObjectMeta{
+			Name: "rule-evaluator",
+		},
+		Spec: appsv1.DeploymentSpec{
+			Replicas: &ruleEvaluatorReplicas,
+		},
+	}
+	emptyRules := newFakeClientBuilder().WithObjects(&alertManager, &ruleEvaluator).Build()
+	hasRule := newFakeClientBuilder().WithObjects(&alertManager, &ruleEvaluator, &monitoringv1.Rules{}).Build()
+	errGettingRules := newFakeClientBuilder().WithInterceptorFuncs(interceptor.Funcs{
+		List: func(_ context.Context, _ client.WithWatch, _ client.ObjectList, _ ...client.ListOption) error {
+			return errors.New("error")
+		}}).Build()
+
```

Comment:
Can you add two unit tests where the rule evaluator/alert manager are missing and see if it still works as expected.